pallets/settlement/src/lib.rs

```diff
@@ -1154,19 +1402,26 @@ impl<T: Trait> Module<T> {
                             SettlementDID.as_id(),
                             instruction_id,
                         ));
+                        // TODO: do we want to emit the error reason as well?
+                        // match reason {
+                        //     FailureReason::BaseTransfer => Err(DispatchError::from(Error::<T>::TODONewError)),
+                        //     FailureReason::ConfidentialTransfer => Err(DispatchError::from(Error::<T>::InvalidMercatTransferProof)),
+                        //     FailureReason::InsufficientMercatAuthorizations => Err(DispatchError::from(Error::<T>::TODONewError)),
+                        //     FailureReason::UndefinedLeg => Err(DispatchError::from(Error::<T>::UndefinedLeg)),
+                        // }
                     }
                 }
             }
             weight_for::weight_for_execute_instruction_if_auth_no_pending::<T>(transaction_weight)
         };
 
-        // Clean up instruction details to reduce chain bloat
+        // Clean up instruction details to reduce chain bloat.
         <InstructionLegs<T>>::remove_prefix(instruction_id);
         <InstructionDetails<T>>::remove(instruction_id);
         <InstructionLegStatus<T>>::remove_prefix(instruction_id);
         <InstructionAuthsPending>::remove(instruction_id);
         <AuthsReceived>::remove_prefix(instruction_id);
```

Comment:
We should also delete the mercat proofs since they are no longer required.

Reply:
hmmm, @ParnianAlimi  do we need to the proofs later on when we work on transaction ordering?

Reply:
As soon as we have the final authorization from the mediator (the last gate keeper), validator can do its checks and delete the proofs. In other words after a transaction has failed or settled we don't need to keep its proofs (we are tracking the transaction amount in an accumulator though).